apps/hash-api/src/index.ts

```diff
@@ -205,13 +218,71 @@ const main = async () => {
   // Potentially only in development mode
   app.use(helmet({ contentSecurityPolicy: false }));
 
+  app.use(express.static("public"));
+
   const jsonParser = json({
     // default is 100kb
     limit: "16mb",
   });
   const rawParser = raw({ type: "application/json" });
 
-  // Body parsing middleware
+  /** PROXIES – these should come BEFORE bodyParser so that the body is proxied without changes */
+
+  /**
+   * Proxy to Ory Hydra's OAuth2 authorization and token endpoints, for OAuth2 clients (e.g. HashGPT)
+   */
+  app.use("/oauth2/auth", rateLimiter, hydraProxy);
+  app.use("/oauth2/token", rateLimiter, hydraProxy);
+  app.use("/oauth2/fallbacks", rateLimiter, hydraProxy);
```

Comment:
We could also put this behind `/auth` to be consistent with how we do the proxy for Ory Kratos, but don't feel super strongly about it if we want to keep these routes simpler. I suppose having them behind `/oauth2` is already consistent in a way.

Reply:
I slightly prefer having these routes which are just a 1:1 to what they are in Hydra (and in many other OAuth2 providers).